BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -293,11 +295,24 @@ def final_task_list #:nodoc:
         if (! f.is_a?( CbrainFileList ) || input.list) # in case of a list input, we *do* assign it the CbFileList
           task = self.dup
           fillTask.( f, task )
-
         else # One task per userfile in the CbrainFileList
-          ufiles = f.userfiles_accessible_by_user!( self.user, nil, nil, file_access_symbol() )
-          # Skip files that are purposefully nil (e.g. given id 0 by the user)
-          subtasks = ufiles.select { |u| ! u.nil? }.map { |u| fillTask.( u, self.dup ) }
+          ufiles               = f.userfiles_accessible_by_user!( self.user, nil, nil, file_access_symbol() )
+          ordered_extra_params = f.is_a?(ExtendedCbrainFileList) ? f.ordered_params : []
+
+          # Remove files that are purposefully nil (e.g. given id 0 by the user)
```

Comment:
Line 302 to 308 seems to be code that identify the null entries in ufiles and remove the entries in the ordered_extra_params accordingly.

But why is this necessary? Can we just have a "next" statement in the loop that's at line 312 instead just to skip over these entries?